### DISCUSSION TITLE
Using Set-RegistryKey to set binary value

### DISCUSSION THREAD

**Discussion by coprolaliant** [PSAppDeployToolkit/PSAppDeployToolkit#866] [Q&A]
I am trying to use the `Set-RegistryKey` cmdlet to set the value of `HKEY_CURRENT_USER\Control Panel\Desktop\UserPreferencesMask` to `90 12 03 80 10 00 00 00`.

This is how I run the cmdlet:

`Set-RegistryKey -Key 'HKEY_CURRENT_USER\Control Panel\Desktop' -Name 'UserPreferencesMask' -Type Binary -Value (0x90,0x12,0x03,0x80,0x10,0x00,0x00,0x00)`

When I check `HKEY_CURRENT_USER\Control Panel\Desktop\UserPreferencesMask` after deployment, its value is `09 00 12 00 30 80 01 00 00 00 00 00`.

What am I doing wrong?

**Comment by jadelise** [comment]
I believe your value as entered is an array which does not store your values in the order you enter them.  Try using an array list, which I belive does or an [ordered] Dictionary object?

**Reply by coprolaliant** [reply]
I will have to do some more testing, but I just tried this and it appeared to work on my computer.

```
$UserPreferencesMaskByteArray = @( 0x90, 0x12, 0x03, 0x80, 0x10, 0x00, 0x00, 0x00 )
$UserPreferencesMaskArrayList = New-Object System.Collections.ArrayList
$UserPreferencesMaskArrayList.AddRange( $UserPreferencesMaskByteArray )
Set-RegistryKey -Key 'HKEY_CURRENT_USER\Control Panel\Desktop' -Name 'UserPreferencesMask' -Type Binary -Value $UserPreferencesMaskArrayList
```

Thank you very much.  I have been struggling with this for a long time.

**Reply by coprolaliant** [reply]
Unfortunately, using an array list does not work.  If I log off the computer and log back on, the incorrect value is restored.

**Reply by mjr4077au** [reply]
@coprolaliant did you ever get this resolved?
